### PR TITLE
Workflow updaterelatedmodules manyto1

### PR DESCRIPTION
**Pull request by loridacito**

### CODE REVIEW

File: modules/com_vtiger_workflow/VTWorkflowUtils.php [hunk 1]
```diff
@@ -68,8 +68,7 @@ public static function fieldvaluebytype($moduleFields, $fieldValueType, $fieldVa
 				}
 			}
 			if ($fieldInstance->getFieldDataType() === 'date') {
-				$date = new DateTimeField($fieldValue);
-				$fieldValue = $date->getDisplayDate();
+				$fieldValue = date("Y-m-d", strtotime($fieldValue));
```
Comment (joebordes) [inline modules/com_vtiger_workflow/VTWorkflowUtils.php]: Are you sure this is correct? That is a very different result. Can we be sure we aren't breaking anything with this change?

I see that the method is only used in the update task and that there, the result will be passed through sanitizeRetrieveEntityInfo. Does that method act correctly when the user has a different setting than Y-m-d?

Comment (loridacito) [reply modules/com_vtiger_workflow/VTWorkflowUtils.php]: It gave me this error while testing it with m-d-y configuration. Here is a var_dump before and after sanitize, without my change in the fiedvaluebytype function:

array (size=22)
  'assigned_user_id' => string '13' (length=2)
  'cyp_no' => string 'PAY-0000091' (length=11)
  'reference' => string 'Elizabeth Ross' (length=14)
  'parent_id' => string '86' (length=2)
  'related_id' => string '3950' (length=4)
  'register' => string '2016-09-14' (length=10)
  'duedate' => string '2016-11-09' (length=10)
  'paymentdate' => string '02-10-2018' (length=10)
  'paid' => string '0' (length=1)
  'credit' => string '1' (length=1)
  'paymentmode' => string 'Transfer' (length=8)
  'paymentcategory' => string 'Infrastructure' (length=14)
  'amount' => string '197.00' (length=6)
  'cost' => string '134.00' (length=6)
  'benefit' => string '63.00' (length=5)
  'createdtime' => string '2015-08-17 04:33:21' (length=19)
  'modifiedtime' => string '2018-02-09 08:49:50' (length=19)
  'reports_to_id' => string '0' (length=1)
  'description' => string 'et nunc. Quisque ornare tortor at risus. Nunc ac sem ut dolor dapibus gravida. Aliquam tincidunt, nunc ac mattis ornare, lectus ante dictum mi, ac mattis velit justo nec ante. Maecenas mi felis, adipiscing fringilla, porttitor vulputate, posuere vulputate, lacus. Cras interdum. Nunc sollicitudin commodo ipsum. Suspendisse non leo. Vivamus nibh dolor, nonummy ac, feugiat non, lobortis quis, pede. Suspendisse dui. Fusce diam nunc, ullamcorper eu, euismod ac, fermentum vel, mauris. Integer' (length=491)
  'created_user_id' => string '1' (length=1)
  'record_id' => string '14384' (length=5)
  'record_module' => string 'CobroPago' (length=9)

After sanitize

array (size=22)
  'assigned_user_id' => string '13' (length=2)
  'cyp_no' => string 'PAY-0000091' (length=11)
  'reference' => string 'Elizabeth Ross' (length=14)
  'parent_id' => string '86' (length=2)
  'related_id' => string '3950' (length=4)
  'register' => string '09-14-2016' (length=10)
  'duedate' => string '11-09-2016' (length=10)
  'paymentdate' => string '10-2018-02' (length=10)
  'paid' => string '0' (length=1)
  'credit' => string '1' (length=1)
  'paymentmode' => string 'Transfer' (length=8)
  'paymentcategory' => string 'Infrastructure' (length=14)
  'amount' => string '197,000000' (length=10)
  'cost' => string '134,000000' (length=10)
  'benefit' => string '63.00' (length=5)
  'createdtime' => string '08-17-2015 06:33' (length=16)
  'modifiedtime' => string '02-09-2018 09:49' (length=16)
  'reports_to_id' => string '0' (length=1)
  'description' => string 'et nunc. Quisque ornare tortor at risus. Nunc ac sem ut dolor dapibus gravida. Aliquam tincidunt, nunc ac mattis ornare, lectus ante dictum mi, ac mattis velit justo nec ante. Maecenas mi felis, adipiscing fringilla, porttitor vulputate, posuere vulputate, lacus. Cras interdum. Nunc sollicitudin commodo ipsum. Suspendisse non leo. Vivamus nibh dolor, nonummy ac, feugiat non, lobortis quis, pede. Suspendisse dui. Fusce diam nunc, ullamcorper eu, euismod ac, fermentum vel, mauris. Integer' (length=491)
  'created_user_id' => string '1' (length=1)
  'record_id' => string '14384' (length=5)
  'record_module' => string 'CobroPago' (length=9)

I tested it now with the d-m-Y format, without my change in the fieldvaluebytype  and it worked correctly. Seems that this error happens only with m-d-Y format.

Could be a problem of the sanitize function. I can investigate . Do you want me to change back the fieldvaluebytype function and than open a new issue or something?

**Comment by evolutivo** [comment]
I investigated the problem by following where the sanitizeRetrieveEntityInfo function is used. It is used in the update ws method, in SaveData.php of Calendar4You in duplicate.php and probably a couple of other places. I tested this three cases and every time the date enters with an Y-m-d format independent of the format of the user.  
So the fix I made in the date part of the fieldvaluebytype function is correct.

**Comment by joebordes** [comment]
this is working very well, I have a few comments that I will add as issues.

Thanks!

**Pull request merged**